platform/windows/detect.py

```diff
@@ -181,6 +181,7 @@ def configure_msvc(env, manual_msvc_config):
 
     if env["tests"]:
         env["windows_subsystem"] = "console"
+        env.AppendUnique(CPPDEFINES=["TESTS_ENABLED"])
```

Comment:
I wonder whether this is actually needed, since running tests without Visual Studio IDE works fine with and without this PR.

It's already defined in `main/SCSub`:

https://github.com/godotengine/godot/blob/cd05653e308c2263e23debd45211c48af134040d/main/SCsub#L13-L14

Reply:
It appears that all other defines are also redefined here, so maybe these CPPDEFINES added in `main/SCSub` is not passed through.

![image](https://user-images.githubusercontent.com/5877844/112161525-8a5f2a00-8beb-11eb-9fd4-6330611bc5b5.png)

Reply:
I'm currently rebuilding the vsproj without that line I'll update once it is finished

Reply:
Yea, if I leave `env.AppendUnique(CPPDEFINES=["TESTS_ENABLED"])` out I don't get the preprocessor definition
![image](https://user-images.githubusercontent.com/5877844/112162038-f6da2900-8beb-11eb-96cb-04ceec29e4ed.png)

Reply:
Yeah, but are you able to run tests without `TESTS_ENABLED` in there?

Reply:
Yep, you're right I can still run the tests even when it is not defined. However, the IDE now thinks it is not going to compile the parts in #ifdefs, and as such makes it look darker. This is not the end of the world, but it would be nice if we could find a way to still "define" it without triggering a full rebuild. Would you know a way to do this?

Reply:
Not sure, I think it's fine, but then the same `TESTS_ENABLED` is better be defined for each platform and not just Windows (for consistency), it's up to @akien-mga to decide I guess.

Reply:
Yeah this shouldn't be added just for IDE's sake, it impacts compilation and that's unnecessary here. This will force rebuilding the whole engine if you change `tests=yes` to `tests=no`, while the current buildsystem would only recompile the files for which it matters (i.e. the sub-environments which are involved in tests).

We could consider changing it and setting `TESTS_ENABLED` in the main env, but that wouldn't be done only for Windows in `detect.py` anyway, and should be considered in a different PR.

Reply:
Okay, I can remove the preprocessor definition, especially seeing as it is not needed to actually run the tests. Are the two other changes (adding the test sources, and adding the tests=yes argument) good to go? I don't have any experience with scons so it could be that these also have unintended consequences that I didn't consider.

Reply:
Yeah the other changes look great :)